src/metabase/api/dashboard.clj

```diff
@@ -829,7 +841,8 @@
     constraint-param-key->value :- ms/Map
     query                       :- [:maybe ms/NonBlankString]]
    (let [constraints (chain-filter-constraints dashboard constraint-param-key->value)
-         field-ids   (param-key->field-ids dashboard param-key)]
+         param       (param-key->param dashboard param-key)
```

Comment:
these 2 functions call `t2/hydrate dashboard :resolved-params)` twice which is redundant, can we somehow reuse it?

Reply:
Do you here under 'redundant' mean that they call it far too many times? It's also done in a loop inside of `chain-filter-constraints`, but I thought it's almost a noop when it's already hydrated (just checked, it takes ~53 µs for me)?

Reply:
> Do you here under 'redundant' mean that they call it far too many times?

yes, but if `dashboard` is not already hydrated when we called `chain-filter`, then it'll be hydrated multiple times in `param-key->param` and `chain-filter-constraints`

especially `chain-filter-constraints` is hydrating in a loop

Reply:
Yeah after looking at it it seems like the best way is to bring hydration up to a `chain-filter` itself.